Realm/RLMSyncUtil_Private.h

```diff
@@ -22,6 +22,17 @@
 #import <Realm/RLMRealmConfiguration.h>
 #import <Realm/RLMSyncCredentials.h>
 
+typedef NS_ENUM(NSUInteger, RLMSyncSystemErrorKind) {
```

Comment:
I just double-checked `asyncOpen`. It can only either return a regular failed-to-open-Realm error, or a sync error created by my `make_sync_error()` helper. We do pass in the error kind, but the function translates it and does other stuff, and doesn't put it into the final `NSError`.